state-sync/storage-service/server/src/tests.rs

```diff
@@ -778,3 +837,169 @@ pub fn initialize_logger() {
         .level(Level::Debug)
         .build();
 }
+
+/// Creates a mock database reader
+pub fn create_mock_db_reader() -> MockDatabaseReader {
```

Comment:
This is super awesome, thank you for adding the real mock functionality for the DB.